worker/src/main/java/org/apache/celeborn/service/deploy/worker/congestcontrol/CongestionController.java

```diff
@@ -122,25 +151,40 @@ public static CongestionController instance() {
    * <p>3. If the pending bytes doesn't exceed the high watermark, will allow all users to try to
    * get max throughout capacity.
    */
-  public boolean isUserCongested(UserIdentifier userIdentifier) {
-    if (userBufferStatuses.size() == 0) {
+  public boolean isUserCongested(UserCongestionControlContext userCongestionControlContext) {
+    if (userBufferStatuses.isEmpty()) {
       return false;
     }
-    if (overHighWatermark.get()) {
-      // If the user produce speed is higher that the avg consume speed, will congest it
-      long userProduceSpeed = getUserProduceSpeed(userBufferStatuses.get(userIdentifier));
-      long avgConsumeSpeed = getPotentialConsumeSpeed();
+
+    UserIdentifier userIdentifier = userCongestionControlContext.getUserIdentifier();
+    long userProduceSpeed = getUserProduceSpeed(userCongestionControlContext.getUserBufferInfo());
+    long avgConsumeSpeed = getPotentialProduceSpeed();
+    // If the user produce speed is higher that the avg consume speed, will congest it
+    if (overHighWatermark.get() && userProduceSpeed > avgConsumeSpeed) {
       if (logger.isDebugEnabled()) {
```

Comment:
maybe compute userProduceSpeed/avgConsumeSpeed after check overHighWatermark

Reply:
Can I do it in the next PR? I will introduce dynamicConfig for CongestionController and support workers to synchronize dynamicConfig from the master.

Reply:
> Can I do it in the next PR? I will introduce dynamicConfig for CongestionController and support workers to synchronize dynamicConfig from the master.

sure, but imo we support dynamic configs same way like master does, it means worker can refresh/use local fs configs/get from db configs.